src/SFML/Graphics/Shader.cpp

```diff
@@ -40,6 +40,18 @@
 
 #ifndef SFML_OPENGL_ES
 
+#if defined(SFML_SYSTEM_MACOS) || defined(SFML_SYSTEM_IOS)
+
+    #define castToGlHandle(x) reinterpret_cast<GLEXT_GLhandle>(static_cast<ptrdiff_t>(x))
+    #define castFromGlHandle(x) static_cast<unsigned int>(reinterpret_cast<ptrdiff_t>(x))
+
+#else
+
+    #define castToGlHandle(x) (x)
+    #define castFromGlHandle(x) (x)
+
+#endif
```

Comment:
Because those lines are literally no-ops. I've never seen any cases where one declares functions just to conditionally cast stuff. Then again not many people run into this problem that often either. :stuck_out_tongue:

What do you suggest?